### PR TITLE
Adding ctre (c++ compile time regular expressions)

### PR DESCRIPTION
**Pull request by faceprint**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by faceprint** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by faceprint** [comment]
Sorry, I didn't realize I needed to do this as a new contributor:

@conda-forge-admin, please ping team

File: recipes/ctre/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (jaimergp) [inline recipes/ctre/build.sh:1]: You probably want error checking:

```suggestion
set -euxo pipefail
```

File: recipes/ctre/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/hanickadot/compile-time-regular-expressions/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  noarch: generic
+
```
Comment (jaimergp) [inline recipes/ctre/meta.yaml:17]:
```suggestion
```

File: recipes/ctre/conda-forge.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+noarch_platforms:
```
Comment (jaimergp) [inline recipes/ctre/conda-forge.yml:1]: This won't have any effects while in staged-recipes but it will work in your feedstock once merged.

**Comment by faceprint** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp PTAL

**Review by xhochy — APPROVED** [review]

**Pull request merged**